src/conversions/std/osstr.rs

```diff
@@ -68,13 +69,11 @@ impl FromPyObject<'_> for OsString {
 
             // Create an OsStr view into the raw bytes from Python
             #[cfg(target_os = "wasi")]
-            let os_str: &OsStr = std::os::wasi::ffi::OsStrExt::from_bytes(
-                fs_encoded_bytes.as_ref(ob.py()).as_bytes(),
-            );
+            let os_str: &OsStr =
+                std::os::wasi::ffi::OsStrExt::from_bytes(fs_encoded_bytes.bind(ob.py()).as_bytes());
```

Comment:
You might be able to use `Py<PyBytes>::as_bytes()` here:

```suggestion
                std::os::wasi::ffi::OsStrExt::from_bytes(fs_encoded_bytes.as_bytes(ob.py()));
```

(Same in the other cfg variant below.)